core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -329,6 +334,23 @@ sealed trait ZIO[-R, +E, +A]
   ): ZIO[R1, E1, A1] =
     catchSomeDefect { case t => h(t) }
 
+  /**
+   * Recovers from all failures with provided function. Equivalent to `catchAll`
+   * but with the provided Cause. If you only care about the error, use
+   * `catchAll` instead.
+   *
+   * This method doesn't recover from defects. If you need to recover from
+   * defects, use `catchAllCause` instead.
+   *
+   * {{{
+   * effect.catchAllFailure(c => ZIO.logErrorCause("failure", c))
+   * }}}
+   */
+  final def catchAllFailure[R1 <: R, E1 >: E, A1 >: A](
```

Comment:
What about naming this `catchFailureCause` instead? I think that would be a more accurate naming for this method right?